### PR TITLE
add test, fixed a bunch of bugs

### PR DESCRIPTION
**Pull request by wanderer**
I apologize, I got carried away did way more than I should in a single commit. The main points are
* fixed several statements
* fixed bug that only allowed an single `generate` call to work

### CODE REVIEW

**Comment by drom** [comment]
@wanderer looks like you have branched old version. Please synchronize before working on the code.

**Comment by drom** [comment]
@wanderer I see that you are using ES2015 features(const, let, template strings, ), 2-space, no-semi, ... style; while original code is ES5, 4-space, var, semi.
I am not against the new trends (where they make sense), but I would love to see all code done in the same style. So, maybe we should set some ground rules?

**Comment by drom** [comment]
@wanderer great work, but looks like you diverging from master.
Please don't return string from each node. Rather concatenate it to the global result.

File: lib/index.js [hunk 1]
```diff
@@ -57,23 +51,14 @@ const exprGen = {
...
   br_table: function (node) { // ['expr', 'body' ],
     let res = '(br_table'
-      // FIXME expr should be an Object
-    node.expr.forEach(function (e) {
-      res += ' ' + exprGen[e.kind](e)
-    })
-    node.body.forEach(function (e) {
-      res += ' ' + exprGen[e.kind](e)
-    })
+    res += node.expr.map((e) => exprGen[e.kind](e)).join(' ')
```
Comment (drom) [inline lib/index.js]: @wanderer `forEach` was there to concatenate to the global result.

File: lib/index.js [hunk 2]
```diff
@@ -30,12 +29,7 @@ const exprGen = {
       `${exprGen[node.right.kind](node.right)})`
   },
   block: function (node) { // ['body' ],
-    let res = '(block'
-    node.body.forEach(function (e) {
-      res += ' ' + exprGen[e.kind](e)
-    })
-    res += ')'
-    return res
+    return `(block ${node.body.map((e) => exprGen[e.kind](e)).join(' ')})`
```
Comment (drom) [inline lib/index.js]: @wanderer I think we should stick with ES5 for now.
Please use `function` instead of `=>`

**Comment by wanderer** [comment]
@drom would an es6 branch ok?  The main branch would be totally isomorphic to it. Just stripping out the es6 semantics. 

>2-space, no-semi, ... style; while original code is ES5, 4-space, var, semi.
uh yeah my text edit formats for me. I was just trying to get a prototype up asap, so i didn't take time to figure out new styling

> Please don't return string from each node
.
will do! I want to add benchmarks first, i'll should have more time to focus on this later this week

**Comment by wanderer** [comment]
ok testing babel, 
this
```
`(block ${node.body.map((e) => exprGen[e.kind](e)).join(' ')})`
```
go to
```
'(block ' + node.body.map(function (e) {
  return exprGen[e.kind](e);
}).join(' ') + ')';
```
which looks fairly good. Might touch up by hand afterwards. 

@drom would you be ok with keeping a copy of es6 around in this repo? or should es6 just be a separate repo altogether?

**Comment by drom** [comment]
**First**. as I mentioned above. IMHO we are not going to use `map` but go back to `forEach` because each node function should append to the `res` string scoped above. We might even go for old good `for` loops for the performance benefits.

**Second**. As we discussed in #5 I don't see any serious benefit using ES6 instead of ES5 for this particular project. Code structure is rather flat and linear. I think it would be very beneficial if we focus on functionality first.

**Comment by drom** [comment]
This PR getting out of any bounds. 100% of code rewritten in completely different language, codding guidelines and tools. I can't see what particular issue it is fixing, or goal in mind. I would prefer more focused and purposeful PRs.

**Comment by wanderer** [comment]
@drom i apologize for oversteping bounds here. I was in a rush and was not considerate.  Ill start over

**Pull request closed without merging**